src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,139 @@ def standardize_patient_name(
     return bundle
 
 
-def standardize_phone(raw: str) -> str:
-    """Make sure it's 10 digits, remove everything else
+def country_extractor(
+    resource: dict, code_type: Literal["alpha_2", "alpha_3", "numeric"] = "alpha_2"
+) -> Generator[str, None, None]:
+    """
+    Given a FHIR resource yield a generator containing all of the counries found in the
+    resource in a standard form sepcified by code_type.
+
+    Args:
+        resource (dict): A FHIR resource respresented as a dictionary.
+        code_type (str): A string equal to 'alpha_2', 'alpha_3', or 'numeric' to specify
+                         which type of standard country identifier to generate.
+    Returns:
+        (generator): A generator to return standardized countries when it is iterated
+                     over.
+
+    We currently only handle extracting countries from patient resources, however as
+    need arises is the future funcionality can be expanded.
+    """
+    resource_type = resource.get("resource").get("resourceType")
+    if resource_type == "Patient":
+        for address in resource.get("resource").get("address"):
+            country = address.get("country")
+            yield standardize_country(country, code_type)
+
 
-    >>> standardize_phone("(555) 555-1212")
-    '5555551212'
+def standardize_phone(
+    raw: str, resource: dict = None, country_extractor: Callable = country_extractor
```

Comment:
Does `country_extractor` need to be passed in like this? Or can we just call it from within `standardize_phone`?

Reply:
I would actually pretty strongly argue we _shouldn't_ be passing functions as `Callable`'s like this. Our principles emphasize composability, and the conversations we've had around how a user should be able to use or not use building blocks to their liking has been of the form 

```
data_1 = building_block_a(raw_data)
data_2 = building_block_c(data_1) # I skipped building_block_b
data_3 = building_block_m(data_2) # skipping BB's c through l
```

That pattern is currently being strongly established in our codebase and in my mind at least it creates some strong benefits around human readability and standardizing what a user can expect. To me it's better to not invoke a callable like this, particular for the smaller case of extracting a country code. In the future, if we do want to add functionality for different kinds of extractors or standardizers, I think those should be given as different functions entirely to keep things discrete, modular, and accessible in bite-sized chunks for non-developer users.